src/Workspaces/CoreTestUtilities/ExportProviderCache.cs

```diff
@@ -138,6 +139,23 @@ public static ComposableCatalog WithPart(this ComposableCatalog catalog, Type t)
             return catalog.WithParts(CreateTypeCatalog(SpecializedCollections.SingletonEnumerable(t)));
         }
 
+        public static ComposableCatalog WithoutPartsOfType(this ComposableCatalog catalog, Type t)
+        {
+            return catalog.WithoutPartsOfTypes(SpecializedCollections.SingletonEnumerable(t));
+        }
+
+        public static ComposableCatalog WithoutPartsOfTypes(this ComposableCatalog catalog, IEnumerable<Type> types)
```

Comment:
Might be worth having a doc comment here that this function includes both types and (being sloppy with terms here) derived types too.